contracts/ronin/gateway/BridgeReward.sol

```diff
@@ -0,0 +1,256 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import "../../extensions/collections/HasContracts.sol";
+import "../../extensions/RONTransferHelper.sol";
+import { IBridgeManager } from "../../interfaces/bridge/IBridgeManager.sol";
+import { IBridgeTracking } from "../../interfaces/bridge/IBridgeTracking.sol";
+import { IBridgeReward } from "../../interfaces/bridge/IBridgeReward.sol";
+import { IBridgeSlash } from "../../interfaces/bridge/IBridgeSlash.sol";
+import { RONTransferHelper } from "../../extensions/RONTransferHelper.sol";
+import { IRoninValidatorSet } from "../../interfaces/validator/IRoninValidatorSet.sol";
+import "../../utils/CommonErrors.sol";
+
+contract BridgeReward is IBridgeReward, HasContracts, Initializable, RONTransferHelper {
+  mapping(address => BridgeRewardInfo) internal _rewardInfo;
+  uint256 internal _rewardPerPeriod;
+  uint256 internal _latestRewardedPeriod;
+
+  constructor() payable {
+    _disableInitializers();
+  }
+
+  function initialize(
+    address bridgeManagerContract,
+    address bridgeTrackingContract,
+    address bridgeSlashContract,
+    address validatorSetContract,
+    uint256 rewardPerPeriod
+  ) external payable initializer {
+    _setContract(ContractType.BRIDGE_MANAGER, bridgeManagerContract);
+    _setContract(ContractType.BRIDGE_TRACKING, bridgeTrackingContract);
+    _setContract(ContractType.BRIDGE_SLASH, bridgeSlashContract);
+    _setContract(ContractType.VALIDATOR, validatorSetContract);
+    _setRewardPerPeriod(rewardPerPeriod);
+  }
+
+  /**
+   * @inheritdoc IBridgeReward
+   */
+  function receiveRON() external payable {}
+
+  function syncReward() external {
+    if (!_isBridgeOperator(msg.sender)) {
+      revert ErrUnauthorizedCall(msg.sig);
+    }
+
+    IBridgeManager bridgeManagerContract = IBridgeManager(getContract(ContractType.BRIDGE_MANAGER));
+    IBridgeTracking bridgeTrackingContract = IBridgeTracking(getContract(ContractType.BRIDGE_TRACKING));
+
+    uint256 currentPeriod = IRoninValidatorSet(getContract(ContractType.VALIDATOR)).currentPeriod();
+    if (currentPeriod <= _latestRewardedPeriod) {
+      revert ErrPeriodAlreadyProcessed(currentPeriod, _latestRewardedPeriod);
+    }
```

Comment:
Since this method is called by the bridge service automatically, should we simply return the method instead of reverting

Reply:
applied.